src/generate-query.ts

```diff
@@ -610,6 +670,121 @@ function getArgsAndVars(
   }
 }
 
+function getListSizeDirective(field: FieldDefinitionNode): DirectiveNode {
+  return field?.directives.find((dir) => dir.name.value === 'listSize')
+}
+
+function getSlicingArguments(listSizeDirective: DirectiveNode): ArgumentNode {
+  if (!listSizeDirective) return undefined
+
+  const slicingArgumentsArg = listSizeDirective.arguments.find(
+    (arg) => arg.name.value === 'slicingArguments'
+  )
+
+  if (!slicingArgumentsArg || slicingArgumentsArg.value.kind !== 'ListValue')
+    return undefined
+
+  return slicingArgumentsArg
+}
+
+function getRequireOneSlicingArgument(
+  listSizeDirective: DirectiveNode
+): ArgumentNode {
+  if (!listSizeDirective) return undefined
+
+  const requireOneSlicingArg = listSizeDirective.arguments.find(
+    (arg) => arg.name.value === 'requireOneSlicingArgument'
+  )
+  if (
+    !requireOneSlicingArg ||
+    requireOneSlicingArg.value.kind !== 'BooleanValue'
+  )
+    return undefined
+
+  return requireOneSlicingArg
+}
+
+function getListSizeDirectiveDefaultValue(
+  listSizeDirective: DirectiveNode,
+  typeNode: TypeNode,
+  config: InternalConfiguration,
+  schema: GraphQLSchema
+) {
+  const slicingArgumentsArg = getSlicingArguments(listSizeDirective)
+  if (!slicingArgumentsArg) return undefined
+  let slicingArgumentsPaths = (slicingArgumentsArg.value as ListValueNode).values.map(
+    (value) => {
+      if (value.kind === 'StringValue') return value.value
+    }
+  )
+  const requireOneSlicingArg = getRequireOneSlicingArgument(listSizeDirective)
+  if (
+    !requireOneSlicingArg ||
+    (requireOneSlicingArg.value as BooleanValueNode).value === true
+  ) {
+    slicingArgumentsPaths = slicingArgumentsPaths.slice(0, 1)
```

Comment:
Why are we slicing here?

Reply:
since require only one slicing is defaulted to true (or if it is present and set to true) then we only can add a single slicing arg in the query / variables. If we have 2 slicing args and that is set to true then its an invalid gql request

Reply:
resolving convo, if you have more questions you can reopen or recomment as needed, thanks